Queueomatic/Server/Endpoints/Login/LoginEndpoint.cs

```diff
@@ -1,32 +1,54 @@
-﻿using FastEndpoints;
+﻿using System.Security.Claims;
+using FastEndpoints;
+using FastEndpoints.Security;
+using Queueomatic.DataAccess.UnitOfWork;
+using Queueomatic.Server.Services.AuthenticationService;
 
 namespace Queueomatic.Server.Endpoints.Login;
 
-public class LoginEndpoint: Endpoint<LoginRequest, LoginResponse>
+public class LoginEndpoint : Endpoint<LoginRequest>
 {
+    private readonly IAuthenticationService _authenticationService;
+    private readonly IConfiguration _configuration;
+    private readonly IUnitOfWork _unitOfWork;
+
+    public LoginEndpoint(IAuthenticationService authenticationService, IConfiguration configuration, IUnitOfWork unitOfWork)
+    {
+        _authenticationService = authenticationService;
+        _configuration = configuration;
+        _unitOfWork = unitOfWork;
+    }
+
+
     public override void Configure()
     {
-        Verbs(Http.POST);
-        Routes("/login");
+        Post("/login");
         AllowAnonymous();
     }
 
     public override async Task HandleAsync(LoginRequest req, CancellationToken ct)
     {
-        try
+        if (!await _authenticationService.CredentialsAreValid(req.Login.Email, req.Login.Password))
         {
-            var response = new LoginResponse();
-            await SendAsync(response, cancellation: ct);
+            await SendUnauthorizedAsync();
```

Comment:
So we will respond with 401 rather 400?

Reply:
Yeah sure. I just thought we wanted just to add an extra security layer.

Can you change it in Wiki?